core/app/models/spree/order.rb

```diff
@@ -516,6 +516,8 @@ def ensure_shipping_address
     end
 
     def create_proposed_shipments
+      return self.shipments if unreturned_exchange? && shipments.all?(&:shipped?)
```

Comment:
I think this should be a raise, possibly even a `raise ... if create_proposed_shipments`. This behaviour doesn't fit well with the rest of this methods behaviour. I think if we were to write documentation for this method it, this behaviour would seem really out of place.

If I change it to this to a `raise if unreturned_exchange? && shipments.all?(&:shipped?)` only the spec testing this exactly this method fails. Is it possible to get a spec which covers _why_ this behaviour is required rather than just that it exists